crates/reward/src/lib.rs

```diff
@@ -64,22 +66,24 @@ pub mod pallet {
     // The pallet's events
     #[pallet::event]
     #[pallet::generate_deposit(pub(crate) fn deposit_event)]
-    // TODO: rename `reward_id` when `pool_id` is added
     pub enum Event<T: Config<I>, I: 'static = ()> {
         DepositStake {
-            reward_id: T::StakeId,
+            pool_id: T::PoolId,
```

Comment:
I know we talked about this before, but I don't quite remember how you answered this previously. Why exactly do we need the `PoolId`? For the capacity pool, it's `()`, and for the normal reward layer it's a `CurrencyId`. However, as far as I can see, the `CurrencyId` we pass is always the vault's collateral currency so it doesn't seem to be doing anything. Note that the reward pallet already supported multiple reward currencies, independently of the vault's collateral currency.

Reply:
For the capacity pool, rewards are distributed based on the stake of each collateral currency which can be defined as `SUM(Collateral[COL] / Threshold[Vault[COL]]) / Price[COL]` - the total minting capacity for the collateral type. This means that rewards are split based on the current exchange rate for each asset.

Let us assume we do not have this `PoolId`, the stake is `Collateral[COL] / Threshold[Vault[COL]]` and the total stake is `SUM(Collateral[COL] / Threshold[Vault[COL]])`. When we distribute rewards now they are split between all Vaults regardless of which collateral currency they are using. Therefore we no longer have need of the capacity pool to differentiate rewards so we lose the context of the exchange rate. This means that two Vaults using different collateral currencies may be able to earn the same amount of rewards providing `Collateral[COL]` and `Threshold[Vault[COL]]` are the same.

Recall that the reason for adding this capacity pool is to simplify the complexity of the solution when the exchange rate is updated. It would be possible to lose the `PoolId` if the stake of each Vault was `Collateral[COL] / Threshold[Vault[COL]]  / Price[COL]` but this means that updates would be `O(n)` since we need to iterate over all Vaults.

Reply:
Maybe I don't understand what you're saying here, but it seems to me that you're talking about not having the capacity reward pool layer, which is not what I am suggesting. What I'm saying is that I think we can have `VaultCapacityRewards` without having a `PoolId` - it's just another instantiation of the Rewards pallet that uses a different `StakeID`